### PR TITLE
JAX: Add padding-and-strides.md

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 1]
```diff
@@ -169,34 +169,65 @@ X = tf.random.uniform(shape=(8, 8))
...
+```{.python .input}
+%%tab jax
+from d2l import jax as d2l
+from flax import linen as nn
+import jax
+from jax import numpy as jnp
+
+# We define a helper function to calculate convolutions. It initializes
+# the convolutional layer weights and performs corresponding dimensionality
+# elevations and reductions on the input and output.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:181]:
```suggestion
# elevations and reductions on the input and output
```

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 2]
```diff
@@ -169,34 +169,65 @@ X = tf.random.uniform(shape=(8, 8))
...
+
 When the height and width of the convolution kernel are different,
 we can make the output and input have the same height and width
 by [**setting different padding numbers for height and width.**]
 
 ```{.python .input}
 %%tab mxnet
-# We use a convolution kernel with height 5 and width 3. The padding on 
+# We use a convolution kernel with height 5 and width 3. The padding on
 # either side of the height and width are 2 and 1, respectively.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:202]:
```suggestion
# either side of the height and width are 2 and 1, respectively
```

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 3]
```diff
@@ -169,34 +169,65 @@ X = tf.random.uniform(shape=(8, 8))
...
 conv2d = nn.Conv2D(1, kernel_size=(5, 3), padding=(2, 1))
 comp_conv2d(conv2d, X).shape
 ```
 
 ```{.python .input}
 %%tab pytorch
-# We use a convolution kernel with height 5 and width 3. The padding on 
+# We use a convolution kernel with height 5 and width 3. The padding on
 # either side of the height and width are 2 and 1, respectively.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:210]:
```suggestion
# either side of the height and width are 2 and 1, respectively
```

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 4]
```diff
@@ -169,34 +169,65 @@ X = tf.random.uniform(shape=(8, 8))
...
 conv2d = nn.LazyConv2d(1, kernel_size=(5, 3), padding=(2, 1))
 comp_conv2d(conv2d, X).shape
 ```
 
 ```{.python .input}
 %%tab tensorflow
-# We use a convolution kernel with height 5 and width 3. The padding on 
+# We use a convolution kernel with height 5 and width 3. The padding on
 # either side of the height and width are 2 and 1, respectively.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:218]:
```suggestion
# either side of the height and width are 2 and 1, respectively
```

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 5]
```diff
@@ -169,34 +169,65 @@ X = tf.random.uniform(shape=(8, 8))
...
 conv2d = tf.keras.layers.Conv2D(1, kernel_size=(5, 3), padding='same')
 comp_conv2d(conv2d, X).shape
 ```
 
+```{.python .input}
+%%tab jax
+# We use a convolution kernel with height 5 and width 3. The padding on
+# either side of the height and width are 2 and 1, respectively.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:226]:
```suggestion
# either side of the height and width are 2 and 1, respectively
```

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 6]
```diff
@@ -135,8 +135,8 @@ comp_conv2d(conv2d, X).shape
 import torch
 from torch import nn
 
-# We define a helper function to calculate convolutions. It initializes 
-# the convolutional layer weights and performs corresponding dimensionality 
+# We define a helper function to calculate convolutions. It initializes
+# the convolutional layer weights and performs corresponding dimensionality
 # elevations and reductions on the input and output.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:140]:
```suggestion
# elevations and reductions on the input and output
```

File: chapter_convolutional-neural-networks/padding-and-strides.md [hunk 7]
```diff
@@ -154,8 +154,8 @@ comp_conv2d(conv2d, X).shape
 %%tab tensorflow
 import tensorflow as tf
 
-# We define a helper function to calculate convolutions. It initializes 
-# the convolutional layer weights and performs corresponding dimensionality 
+# We define a helper function to calculate convolutions. It initializes
+# the convolutional layer weights and performs corresponding dimensionality
 # elevations and reductions on the input and output.
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/padding-and-strides.md:159]:
```suggestion
# elevations and reductions on the input and output
```
